src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -482,6 +485,11 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
 
         }
 
+        private static uint GetValEscape(TypeSymbol type, uint possibleValEscape)
```

Comment:
I think you need some justification here. Not all non-ref variable references have the same escape scope as the local scope, so you need some reason why this will always be true.

Reply:
A value whose type is not a ref-struct type can escape everywhere. The purpose of this method is just to implement this one rule.